src/lib.rs

```diff
@@ -1219,6 +1301,70 @@ pub fn handle_compile_subcommand(
     Ok(json!({"descriptor": descriptor.to_string()}))
 }
 
+/// Proof of reserves verification sub-command
+///
+/// Proof of reserves options are described in [`CliSubCommand::Reserves`].
+#[cfg(all(feature = "reserves", feature = "electrum"))]
+pub fn handle_reserves_subcommand(
+    network: Network,
+    message: String,
+    addresses: Vec<String>,
+    psbt: String,
+    electrum_opts: ElectrumOpts,
+) -> Result<serde_json::Value, Error> {
+    let psbt = base64::decode(&psbt).unwrap();
+    let psbt: PartiallySignedTransaction = deserialize(&psbt).unwrap();
+    let client = Client::new(&electrum_opts.server).unwrap();
+
+    let outpoints_per_addr: Result<Vec<_>, _> = addresses
+        .iter()
+        .map(|address| {
+            let address = Address::from_str(&address).unwrap();
+            get_outpoints_for_address(address, &client)
+        })
+        .collect();
+    let outpoints_combined = outpoints_per_addr?
```

Comment:
It's more clear to unwrap the the result in the previous step than before `iter()`.

Reply:
I agree, that when reading the code, it would look more natural. 
But If I just move the "?" operator up, then the automatic type inference does not work as nicely, and neither the automatic abortion when hitting an item with Err. I experimented a while with that algo, and this is the cleanest that I found. It nicely makes use of moving the Err from any item out to the whole collection: https://doc.rust-lang.org/std/result/enum.Result.html#method.from_iter
Somehow I couldn't get that to work without a ton of type annotations when moving the ? operator to the first part.

Reply:
You can use something like
```
    let outpoints_per_addr = addresses
        .iter()
        .map(|address| {
            let address = Address::from_str(&address)
                .map_err(|_| Error::Generic("Invalid address".to_string()))?;
            Ok(get_outpoints_for_address(address, &client)?)
        })
        .collect::<Result<Vec<Vec<_>>, Error>>()?;
```

rust automatically converts `Vec<Result<_>>` into `Result<Vec<_>>` by simple type declaration after `collect`. That can be used to unwrap the whole collection and also propagate the error.